### ISSUE TITLE
array_join ignores internal NULL except when trailing

### ISSUE THREAD

**Issue by MichaelChirico** [trinodb/trino#4487]
```
select array_join(array[NULL, '1', NULL, '2', NULL], ', ')
-- 1, 2,
```

Neither the first nor the second `NULL` mattered for output, but the last one led `, ` to be inserted. This seems inconsistent.

It seems I've got to do an extra `filter` step to get my desired output:

```
select array_join(filter(array[NULL, '1', NULL, '2', NULL, NULL], x -> x is not null), ', ')
-- 1, 2
```

**Comment by dain** [comment]
I'll fix this. Found this the other day and already have a fix in progress.

**Comment by stubartmess** [comment]
We were pretty baffled by this behavior today, when different runs resulted in different numbers of delimiters. Is the fix still in progress @dain ?

**Comment by tangjiangling** [comment]
@dain Is this fix done yet, if not I want to try it ~

**Comment by tangjiangling** [comment]
I found that this issue has been fixed by @dain in [f04fa5ca](https://github.com/trinodb/trino/commit/f04fa5ca8de1015629a94693740dfbc641e55784), and to avoid causing disruption, this issue can be closed!
cc @findepi

**Comment by hashhar** [comment]
Fixed in https://github.com/trinodb/trino/pull/4343 with tests (https://github.com/trinodb/trino/pull/4343/files#diff-7f28018bb297e0fa90001493f75cc9f6b7a7828f3ce35100d63072ea0cb4d481R536).

Closing. Thanks for tracking this down @tangjiangling .
